src/index.ts

```diff
@@ -1,61 +1,148 @@
-import { Reshuffle, BaseConnector, EventConfiguration } from 'reshuffle-base-connector'
+import {
+  BaseHttpConnector,
+  EventConfiguration,
+  Reshuffle,
+} from '@reshuffle/base-connector'
+import {
+  parseFilter,
+  unwrapDates,
+  wrapDates,
+} from "./util"
 
-export interface _CONNECTOR_NAME_ConnectorConfigOptions {
-  var1: string
-  // ...
+import {Request, Response} from 'express'
+
+const DEFAULT_WEBHOOK_PATH = '/webhooks/wix'
+
+export interface WixConnectorConfigOptions {
+  secret?: string,
+  webhookPath: string,
+}
+
+export type WixAction =
+  'provision'
+  | 'schemas/find'
+  | 'schemas/list'
+  | 'data/get'
+  | 'data/count'
+  | 'data/find'
+  | 'data/insert'
+  | 'data/update'
+
+export interface WixConnectorEventOptions {
+  action: WixAction
 }
 
-export interface _CONNECTOR_NAME_ConnectorEventOptions {
-  option1?: string
-  // ...
+export interface WixRequestContext {
+  settings: Record<string, any>
+  instanceId: string
+  installationId: string
+  memberId: string
+  role: string
 }
 
-export default class _CONNECTOR_NAME_Connector extends BaseConnector<
-  _CONNECTOR_NAME_ConnectorConfigOptions,
-  _CONNECTOR_NAME_ConnectorEventOptions
-> {
-  // Your class variables
-  var1: string
+export interface WixEvent {
+  requestContext: WixRequestContext
+  collectionName?: string
+  filter?: string
+  sort?: any
+  skip?: number
+  limit?: number
+  itemId?: string
+  item?: any
+  body?: any
+  action: WixAction
+  request: Request
+  response: Response
+}
+
+class WixConnector extends BaseHttpConnector<WixConnectorConfigOptions,
+  WixConnectorEventOptions> {
+
+  private readonly secret: string | undefined
```

Comment:
- secret doesn't have to be kept in class. you can always access it with this.configOptions.secret
- webhookUrl isn't used anywhere outside the onStart, no need to have it as class variable

Reply:
Great points! Thanks

Reply:
The `webhookUrl` (now renamed to `webhookPath`) is used to extract the action (See `extractAction`)